writeengine/bulk/we_bulkloadbuffer.cpp

```diff
@@ -1721,107 +1690,1381 @@ int BulkLoadBuffer::parseCol(ColumnInfo& columnInfo)
       }
     }
 
-    if (bufStats.satCount)  // @bug 3504: increment row saturation count
+    if (bufStats.satCount)
     {
-      // If we don't want to allow saturated values for auto inc columns.
-      // then this is where we handle it.  Too late to reject a single
-      // row from the parsing thread, so we abort the job.
-      // if (columnInfo.column.autoIncFlag)
-      //{
-      //    rc = ERR_AUTOINC_USER_OUT_OF_RANGE;
-      //    WErrorCodes ec;
-      //    ostringstream oss;
-      //    oss << "parseCol: error with auto-increment values "
-      //        "for table-" << fTableName <<
-      //        ", column-" << columnInfo.column.colName <<
-      //        "; OID-" << columnInfo.column.mapOid <<
-      //        "; " << ec.errorString(rc);
-      //    fLog->logMsg( oss.str(), rc, MSGLVL_ERROR );
-      //    return rc;
-      //}
       columnInfo.incSaturatedCnt(bufStats.satCount);
     }
 
-    delete[] field;
-    section->write(buf, fTotalReadRowsParser);
+    section->write(buf, numRows);
     delete[] buf;
-#ifdef PROFILE
-    Stats::stopParseEvent(WE_STATS_PARSE_COL);
-#endif
-
-    // TODO MCOL-641 Add support here.
-    if (fLog->isDebug(DEBUG_2))
-    {
-      ostringstream oss;
-      RID rid1 = section->startRowId();
-      RID rid2 = section->endRowId();
-      oss << "ColRelSecOut: OID-" << columnInfo.column.mapOid << "; StartRID/Rows2: " << rid1 << " "
-          << (rid2 - rid1) + 1 << "; startOffset: " << section->getStartOffset()
-          << "; lastExtentRow: " << lastInputRowInExtent;
-      parseColLogMinMax(oss, columnInfo.column.dataType, bufStats.minBufferVal, bufStats.maxBufferVal);
-
-      fLog->logMsg(oss.str(), MSGLVL_INFO2);
-    }
 
     RETURN_ON_ERROR(columnInfo.fColBufferMgr->releaseSection(section));
   }
-
   return rc;
 }
```

Comment:
I think you can keep the corner case after the `for` loop in the original `parseCol()` function, in `parseColParquet()`, instead of moving it inside `convertParquet()`. This maintains the same consistency as the original function.